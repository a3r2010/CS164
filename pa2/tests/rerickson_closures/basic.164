def ryan(a,b,c) {
  print a
  print b
  if(c == 0) {
    print "Base case!"
    def ryan(b) {
      def c = 0
      print "It's over..."
      print a + b + c
    }
    a = 9000
    c = -9000
    ryan(0)
  } else {
    ryan(b, a, c-1)
  }
}

ryan(2112,1337,2)