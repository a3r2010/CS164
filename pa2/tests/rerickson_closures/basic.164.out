2112
1337
1337
2112
2112
1337
Base case!
It's over...
9000
